src/implicit_function.jl

```diff
@@ -15,21 +33,62 @@ This amounts to solving a linear system `A * J = -B`, where `A ∈ ℝᵈˣᵈ`,
 - `conditions::C`: callable of the form `(x,y,z) -> F(x,y,z)`
 - `linear_solver::L`: callable of the form `(A,b) -> u` such that `Au = b`, must be taken from Krylov.jl
 """
-struct ImplicitFunction{F,C,L}
+struct ImplicitFunction{F<:Forward,C<:Conditions,L}
     forward::F
     conditions::C
     linear_solver::L
 end
 
 """
-    ImplicitFunction(forward, conditions)
+    ImplicitFunction(forward, conditions, linear_solver, ::Val{returns_byproduct} = Val(false))
 
-Construct an `ImplicitFunction` with `Krylov.gmres` as the default linear solver.
+Construct an `ImplicitFunction` with `linear_solver` as the linear solver used in the implicit
+differentiation. By default, `ImplicitFunction(forward, conditions, linear_solver)` will assume
+that the `forward` function returns a single output and that the `conditions` function accepts
+two arguments.
```

Comment:
Name them to get the order right: `x` and `y`

Reply:
not sure what you mean here

Reply:
Just specifying the order in which arguments must be given, but I guess it's fine